<a href="https://colab.research.google.com/github/chaitanysaraswat/Advanced-Analytics-Using-Statistics/blob/main/lecture15.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Case1:
simple linear regression:

- Single predictor single response both continuous

- graph :scatterplot

- tools:corelation coeficient

- null hypothesis: predictor does not have influence on response

- if pvalue is less than los we reject null hypothesis

- predictor model: simple linear regression

# Case2:
multiple linear regression:
- multiple predictor(all continuous) single response(continuous) both continuous

- graph :multiple scatterplot

- tools:corelation coeficient

r squared vs adj r squared willbe lower there will be penality for no of predictor

In [ ]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols
import statsmodels.stats.multicomp
import sklearn
from sklearn.model_selection import train_test_split
import numpy as np
import pandas as pd
import warnings
warnings.filterwarnings("ignore")
pd.set_option('display.max_columns', None)
from numpy import random
from collections import Counter
from numpy.linalg import inv
from numpy.linalg import eig
import matplotlib
from matplotlib import pyplot as plt
import seaborn as sns
import pylab
from pylab import legend
from pylab import plot, show, title, xlabel, ylabel
import scipy
from scipy import stats
from scipy.stats import binom,poisson,norm,t,expon,f
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_squared_error

In [ ]:
from sklearn.linear_model import LinearRegression
import pandas as pd
from sklearn.model_selection import train_test_split
df = pd.read_excel('/content/drive/MyDrive/My pitch/CDAC_DataBook.xlsx', sheet_name = 'stackloss')
x_train, x_test, y_train, y_test = train_test_split(df.drop('StackLoss', axis=1),df.StackLoss, test_size=0.1, random_state=20)

# Creating an instance for the function
regr1 = LinearRegression()

regr1.fit(x_train,y_train)

print(regr1.intercept_)
print(regr1.coef_)
print(regr1.feature_names_in_)

-33.281344575611186
[ 0.70013521  1.34985785 -0.22974125]
['AirFlow' 'WaterTemp' 'AcidConc']


# **`For Standardization of linear regression`**

when the values are standardized distribution of values on histogram will not

change only mean=0 and std=1  

In [ ]:
from sklearn.preprocessing import StandardScaler
scl = StandardScaler()

x_train = scl.fit_transform(x_train)
x_test = scl.transform(x_test) # we transform not transform fit becoz we don't want any information to be leaked to
regr2 = LinearRegression()
regr2.fit(x_train,y_train)

regr2.intercept_   # intercept
regr2.coef_    # coefficients

array([ 6.19332814,  4.40861726, -0.90286981])

In [ ]:
from sklearn.preprocessing import StandardScaler
scl = StandardScaler()
x_train, x_test, y_train, y_test = train_test_split(df.drop('StackLoss', axis=1),df.StackLoss, test_size=0.1, random_state=20)
x_train = scl.fit_transform(x_train)
x_test = scl.transform(x_test)

x_train = pd.DataFrame(x_train,columns=['AirFlow','WaterTemp','AcidConc'])
x_test = pd.DataFrame(x_test,columns=['AirFlow','WaterTemp','AcidConc'])
regr2 = LinearRegression()
regr2.fit(x_train,y_train)

print(regr2.intercept_)   # intercept
print(regr2.coef_)    # coefficients
print(regr2.feature_names_in_)

18.666666666666668
[ 6.19332814  4.40861726 -0.90286981]
['AirFlow' 'WaterTemp' 'AcidConc']


# `Feature Scaling`
## Scaling Technique
- standarized
formula=orginal_values-mean/std

## very large variable get effect on model  thats why we do standardizationn


In [ ]:
df = pd.DataFrame({'Values':[3,6,5,1,6,8,3,8,9,4]})

df['StdScale'] = (df.Values-np.mean(df.Values))/np.std(df.Values,ddof=1)

In [ ]:
df


,Values,StdScale
0,3,-0.890045
1,6,0.270883
2,5,-0.116093
3,1,-1.663997
4,6,0.270883
5,8,1.044835
6,3,-0.890045
7,8,1.044835
8,9,1.431811
9,4,-0.503069


## Min-Max scaler
- Min-Max Scaler: (x - min(col)) / (max(col) - min(value))

- cons:this can be affected by outlier

In [ ]:
df['MinMax']=(df.Values-min(df.Values))/(max(df.Values)-min(df.Values))
df

,Values,StdScale,MinMax
0,3,-0.890045,0.250
1,6,0.270883,0.625
2,5,-0.116093,0.500
3,1,-1.663997,0.000
4,6,0.270883,0.625
5,8,1.044835,0.875
6,3,-0.890045,0.250
7,8,1.044835,0.875
8,9,1.431811,1.000
9,4,-0.503069,0.375


## Robust Scaler :
- (x-median(col))/percentile(col,75)-percentile((col,75))

- if specially distance model like knn,lknmeans,pca this scaling down important it can be ok in regression there is no need in decission tree.

In [ ]:
df['Robust'] = (df.Values-np.median(df.Values))/(np.percentile(df.Values,75)-np.percentile(df.Values,25))
df

,Values,StdScale,MinMax,Robust
0,3,-0.890045,0.250,-0.588235
1,6,0.270883,0.625,0.117647
2,5,-0.116093,0.500,-0.117647
3,1,-1.663997,0.000,-1.058824
4,6,0.270883,0.625,0.117647
5,8,1.044835,0.875,0.588235
6,3,-0.890045,0.250,-0.588235
7,8,1.044835,0.875,0.588235
8,9,1.431811,1.000,0.823529
9,4,-0.503069,0.375,-0.352941


# case 3:

Response continuous predictors atleast one categorical and others can be continuous.

c coefficientis zeo
p value 0.33
doen not reject
c does not increase it i

In [ ]:
df = pd.read_excel('/content/drive/MyDrive/My pitch/CDAC_DataBook.xlsx', sheet_name = 'salaries')

df = df[['yrs_service','gender','salary']]

gender_dum = pd.get_dummies(df.gender,drop_first=True).astype(int)

df = pd.concat([df,gender_dum], axis=1)

df = df.drop('gender', axis=1)


# Split for train-test
x_train, x_test, y_train, y_test = train_test_split(df.drop('salary', axis=1),df.salary, test_size=0.2, random_state=20)

# Add constant term
x_train = sm.add_constant(x_train, prepend=False)
x_test = sm.add_constant(x_test, prepend=False)

# Prepare the model
mod1 = sm.OLS(y_train,x_train).fit()

print(mod1.summary())

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.121
Model:                            OLS   Adj. R-squared:                  0.115
Method:                 Least Squares   F-statistic:                     21.52
Date:                Thu, 04 Jun 2026   Prob (F-statistic):           1.75e-09
Time:                        11:46:25   Log-Likelihood:                -3698.4
No. Observations:                 317   AIC:                             7403.
Df Residuals:                     314   BIC:                             7414.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yrs_service   701.6382    125.946      5.571      

In [ ]:
df.head()

,yrs_service,salary,Male
0,18,139750,1
1,16,173200,1
2,3,79750,1
3,39,115000,1
4,41,141500,1


null hypothesis for above:
- w.r.t female if a male is selected than it does not influence salary
- if you cant understand say gender doent have nukk hypothesis

### step2:p value 0.012
### step3 we reject null hypothesis
### step4 significant impact


for the constant null hypothesis

step1: c=0

step2: p=0

since value is 0.05 we reject null

step3: constant term influences salary

const8.857 indicates the salary of person for whom yrs_exp=0 and male =0 which means that the gender is female

the starting salary for male would be (8.85 +1.368)salary

In [ ]:
df = pd.read_excel('/content/drive/MyDrive/CDAC_stats/CDAC_DataBook.xlsx', sheet_name = 'salaries')

rank = df[['yrs_service','rank','salary']]

rank_dum = pd.get_dummies(df.rank,drop_first=True).astype(int)

df = pd.concat([df,rank_dum], axis=1)

df = df.drop('gender', axis=1)


# Split for train-test
x_train, x_test, y_train, y_test = train_test_split(df.drop('salary', axis=1),df.salary, test_size=0.2, random_state=20)

# Add constant term
x_train = sm.add_constant(x_train, prepend=False)
x_test = sm.add_constant(x_test, prepend=False)

# Prepare the model
mod1 = sm.OLS(y_train,x_train).fit()

print(mod1.summary())

TypeError: object of type 'method' has no len()

CONSTANT TERM INDICATES WHEN CONTINUOUS PREDICTOR ARE EQUAL TO ZERO  AND THE CATEGORICAL VARIABLES ARE SET AT REFERENCE LEVEL

In [ ]:
df = pd.read_excel('/content/drive/MyDrive/CDAC_stats/SalesData.xlsx')


region_dum = pd.get_dummies(df['Region'],drop_first=True).astype(int)

df = pd.concat([df,region_dum], axis=1)

df = df.drop('Region', axis=1)


# Split for train-test
x_train, x_test, y_train, y_test = train_test_split(df.drop('Sales', axis=1),df.Sales, test_size=0.2, random_state=20)

# Add constant term
x_train = sm.add_constant(x_train, prepend=False)
x_test = sm.add_constant(x_test, prepend=False)

# Prepare the model
mod1 = sm.OLS(y_train,x_train).fit()

print(mod1.summary())
# EQUATION FOR MODEL
# sales=3.5*budget0.02*store_size-4.08*north+10.8*south-5.9*west-68.6

                            OLS Regression Results                            
Dep. Variable:                  Sales   R-squared:                       0.888
Model:                            OLS   Adj. R-squared:                  0.887
Method:                 Least Squares   F-statistic:                     1889.
Date:                Thu, 04 Jun 2026   Prob (F-statistic):               0.00
Time:                        12:26:09   Log-Likelihood:                -5940.0
No. Observations:                1200   AIC:                         1.189e+04
Df Residuals:                    1194   BIC:                         1.192e+04
Df Model:                           5                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Ad_Budget      3.4866      0.037     93.243      0.0

Instructions:
For each situation, identify the most appropriate inferential statistical test. Do not perform any calculations. Simply decide which statistical tool should be used.


---

Question 1

A car manufacturer claims that the average fuel efficiency of a particular model is 22 km/liter. A sample of 50 cars is tested to verify the claim.
1 sample z test

---

Question 2

A company wants to compare the average monthly sales generated by employees in the North region and the South region.

2 sample t test
---

Question 3

A university compares students' examination scores before and after attending a special coaching program.
paired t test

---

Question 4

A hospital wants to compare the average recovery time among patients receiving three different treatments.

one way anova
---

Question 5

A restaurant owner believes that 70% of customers prefer online ordering. A survey of customers is conducted to verify this claim.

1 prop t test
---

Question 6

A researcher wants to determine whether smoking status (Smoker/Non-Smoker) is related to the occurrence of lung disease (Yes/No).

degree of assosciation
---

Question 7

A manufacturing company studies whether machine type (A, B, C) and operator experience level (Junior/Senior) influence production output.

degree of association
---

Question 8

A political analyst wants to determine whether support for a candidate differs from 50% in a particular city.
1 prop t-test

---

Question 9

An e-commerce company compares the percentage of customers who purchase a product after seeing Advertisement A versus Advertisement B.

2 prop t test
---

Question 10

A school principal wants to compare average mathematics scores among students from four different schools.

one way anova
---

Question 11

A pharmaceutical company compares the average cholesterol levels of patients taking Drug A and Drug B.
 2 sample t-test

---

Question 12

A quality inspector knows that the population standard deviation of a component's weight is 1.5 grams. A sample is collected to determine whether the average weight differs from 50 grams.

1 sample t-test
---

Question 13

A retailer expects 25% of customers to buy Electronics, 35% Clothing, 20% Home Appliances, and 20% Groceries. Actual purchase data is collected to check whether this distribution holds.

goodness of fit
---

Question 14

A company measures employee productivity before and after introducing a new software system.
paired t test
---

Question 15

A bank wants to know whether loan approval status is associated with marital status.
degree of assosciation

---

Question 16

An airline wants to compare average customer satisfaction scores among passengers traveling in Economy, Premium Economy, Business, and First Class.

one way anova
---

Question 17

A beverage company compares the average daily sales of a product in two different cities.
2 sample t test

---

Question 18

A hospital compares the percentage of patients who recover after receiving Treatment A and Treatment B.
2 proportion test

---

Question 19

A researcher investigates whether education level and voting preference are related.

degree of association
---

Question 20

A company studies whether average employee performance differs across five departments.

one anova
---

Question 21

A fitness center records members' body fat percentages before and after a 12-week fitness program.
paired t test

---

Question 22

A manufacturer wants to determine whether the average diameter of produced bearings differs from the specified standard of 20 mm.

1 sample t test
---

Question 23

A marketing team studies whether both advertising channel (TV, Social Media, Print) and region (North, South) influence sales revenue.

degree of assosciation
---

Question 24

A telecom company claims that 95% of customer complaints are resolved within 48 hours. A sample is taken to evaluate the claim.

1 poisson
---

Question 25

A researcher compares the percentage of male and female students who pass a certification examination.

degree of assosciation/ 2 proportion
---

Question 26

A casino wants to determine whether a roulette wheel produces outcomes according to the expected probabilities.
goodness of fit

---

Question 27

A company compares average salaries among employees with Bachelor's, Master's, and PhD degrees.

one way anova
---

Question 28

A hospital compares average blood sugar levels of diabetic and non-diabetic patients.

2 sample t test
---

Question 29

A researcher investigates whether customer age group and preferred payment method are associated.
degree of assosciation

---

Question 30

A manufacturing firm studies whether production shift (Morning, Evening, Night) and machine type jointly affect defect rates.
degree of assosciation

---

Question 31

A sports academy measures athletes' sprint times before and after a specialized training camp.
paired t test

---

Question 32

A mobile phone company wants to verify whether the average battery life of a model is 18 hours as advertised.

 1 sample t test
---

Question 33

A retailer compares the conversion rates of two different website designs.

2 sample poisson test
---

Question 34

A university wants to determine whether average GPA differs among students enrolled in Engineering, Commerce, Arts, and Science programs.

one way anova
---

Question 35

A survey records respondents' preferred mode of transportation (Car, Bus, Metro, Bike). The city authority wants to know whether the observed preferences match previously reported proportions.

goodness of fit
---

Question 36

A company compares average customer waiting times at two service centers.

2 sample t
---

Question 37

A researcher investigates whether employment status and home ownership are related.

degree of asssociation
---

Question 38

A school wants to evaluate whether a new teaching method improves students' science scores by comparing performance before and after implementation.

paired t test
---

Question 39

A food delivery company compares the proportion of on-time deliveries between two delivery partners.
2 proportion

---

Question 40

A researcher studies whether average crop yield differs across six fertilizer types.

one way anova